compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -370,6 +381,7 @@ class ComposeScene internal constructor(
         ) {
             CompositionLocalProvider(
                 LocalComposeScene provides this,
+                LocalPointerPositions provides pointerPositions,
```

Comment:
Ok, just to clarify what we're doing here.
`LocalPointerPositions` provides a static value of a `MutableMap` that isn't a snapshot map. So you can't use this to track the pointers, only to obtain the "current" pointer positions.

Reply:
Do we even need to provide this as a `CompositionLocal`? It's not something that is going to change in the composition hierarchy, and it possibly creates the wrong impression that these coordinates are local to the element where they are consumed.

Maybe instead use
```
LocalComposeScene.current.pointerPositionsInWindow
```
avoiding the extra `CompositionLocal`, and clarifying that these are the positions relative to the window.

Reply:
> So you can't use this to track the pointers, only to obtain the "current" pointer positions.

Yes, this is what I wanted to achieve.

> coordinates are local to the element where they are consumed

Really good point! We have discussed it with @igordmn before implementation and thought that CompositionLocal is better because we don't stuck on ComposeScene then. But I am fully agree with your argument and I think it is better to change CompositionLocal to internal state of ComposeScene
@igordmn WDYT?

Reply:
Moved to internal property

Reply:
> LocalComposeScene.current.pointerPositionsInWindow

The reason why I didn't recommend it, because we become tightly coupled with ComposeScene (so, some low-level component access high-level component).

To resolve this, we extract a mediator that is used by a low-level component and high-level ComposeScene. It can be either `LocalCurrentPointers`, or something like `LocalWindowInfo.currentPointers` (not LocalWindow.currentPointers).

Because we changed it to LocalComposeScene and it is internal, it is fine right now. Otherwise, if it was public, preferreable put to it into `LocalWindowInfo`.